### PR TITLE
Use colcon instead of ament_tools

### LINKED ISSUE #21: Replace ament_tools with colcon

### PR DESCRIPTION
**Pull request by esteve**
This just replaces the use of ament_tools in the README and the CI scripts with colcon

Closes #21

### CODE REVIEW

**Comment by esteve** [comment]
@adamdbrw @theseankelly could you have a look at this to check that it works for you? Thanks.

File: README.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ md \dev\ros2_dotnet_ws\src
 cd \dev\ros2_dotnet_ws
 curl -sk https://raw.githubusercontent.com/esteve/ros2_dotnet/master/ros2_dotnet.repos -o ros2_dotnet.repos
 vcs import \dev\ros2_dotnet_ws\src < ros2_dotnet.repos
-src\ament\ament_tools\scripts\ament.py build
+colcon build --merge-install
```
Comment (theseankelly) [inline README.md]: FYI the necessary fixes to `colcon-cmake` (for the -A flag) are *not* included in the latest colcon meta-release. May want to add a note that the user should ensure this subpackage is version 0.2.12 or higher.

File: ci/build-all-steps.yml [hunk 2]
```diff
@@ -84,14 +88,14 @@ steps:
       call install\local_setup.bat
       cd \dev\ros2
       vcs branch
-      ament build ^
-        --parallel ^
+      colcon build ^
+        --merge-install ^
         --cmake-args ^
-          -G "${{ parameters.generator }}" ^
```
Comment (theseankelly) [inline ci/build-all-steps.yml]: Looks like colcon-cmake also now supports "Visual Studio 16 2019" so you might as well leave this parameter in for the future

File: ci/build-all-steps.yml [hunk 3]
```diff
@@ -84,14 +88,14 @@ steps:
...
+          -A "${{ parameters.arch }}" ^
+          -DBUILD_TESTING=OFF ^
           -DTHIRDPARTY=ON ^
           -DINSTALL_EXAMPLES=OFF ^
-          -DCMAKE_FIND_ROOT_PATH="\dev\ament\install;\dev\ros2\install" ^
-        --
+          -DCMAKE_FIND_ROOT_PATH="\dev\ament\install;\dev\ros2\install"
```
Comment (theseankelly) [inline ci/build-all-steps.yml]: I think this shouldn't be necessary with colcon. I haven't needed it so far, at least.

File: README.md [hunk 4]
```diff
@@ -97,7 +97,7 @@ vcs import src < ros2_dotnet_uwp.repos
 cd \dev\ament
 call install\local_setup.bat
 cd \dev\ros2
-ament build --parallel --cmake-args -G "%TARGET_GENERATOR%" -DCMAKE_SYSTEM_NAME=WindowsStore -DCMAKE_SYSTEM_VERSION=10.0.14393 -DTHIRDPARTY=ON -DINSTALL_EXAMPLES=OFF -DCMAKE_FIND_ROOT_PATH="\dev\ament\install;\dev\ros2\install" --
+colcon build --merge-install --cmake-args -A "%TARGET_ARCH%" -DCMAKE_SYSTEM_NAME=WindowsStore -DCMAKE_SYSTEM_VERSION=10.0.14393 -DTHIRDPARTY=ON -DINSTALL_EXAMPLES=OFF -DCMAKE_FIND_ROOT_PATH="\dev\ament\install;\dev\ros2\install"
```
Comment (theseankelly) [inline README.md]: -DCMAKE_FIND_ROOT_PATH is not necessary (same as comment below)

Comment (theseankelly) [reply README.md]: also need to include `-DBUILD_TESTING=OFF`

File: ci/build-all-steps.yml (hunk 2 above)
Comment (esteve) [reply ci/build-all-steps.yml]: I've added this back, but unfortunately if the `VisualStudioVersion` environment variable is not present, `colcon` will fail

File: ci/build-all-steps.yml (hunk 3 above)
Comment (esteve) [reply ci/build-all-steps.yml]: Removed

File: README.md (hunk 4 above)
Comment (esteve) [reply README.md]: Added.

File: README.md (hunk 1 above)
Comment (esteve) [reply README.md]: That's strange, I added `colcon` via `colcon-common-extensions` to the CI and it seems to have accepted the `-A` flag, do you think the instructions should have a call to `pip install colcon-cmake>=0.2.12`?

**Pull request merged**